deepwell/src/services/filter/service.rs

```diff
@@ -91,6 +99,32 @@ impl FilterService {
             ..Default::default()
         };
 
+        // Handle case-sensitivity logic
+        if let ProvidedValue::Set(case_sensitive) = case_sensitive {
+            match regex {
+                // If the regex is being changed, add case-insensitivity flag if case-insensitive.
+                ProvidedValue::Set(ref mut regex) if !case_sensitive => {
+                    regex.insert_str(0, "(?i)")
+                }
+
+                // If the regex is not being changed, add/remove the flag from the database regex.
+                ProvidedValue::Unset => {
+                    let mut model_regex =
+                        model.get(filter::Column::Regex).as_ref().to_string();
+
+                    if case_sensitive {
+                        model_regex = str!(model_regex.trim_start_matches("(?i)"));
+                    } else {
+                        model_regex.insert_str(0, "(?i)");
```

Comment:
Good catch, since an update to the case sensitivity doesn't necessarily imply it is changing the case sensitivity, just that a new value was provided. Will fix.